magda-gateway/src/index.ts

```diff
@@ -228,6 +230,15 @@ const authenticator = new Authenticator({
     dbPort: argv.dbPort
 });
 
+const startIndex = argv.externalUrl.indexOf("//");
```

Comment:
Use urijs to parse the host out of the URL.

Reply:
done.